### DISCUSSION TITLE
Autocompletions for file paths in Notebooks

### DISCUSSION THREAD

**Discussion by tanhakabir** [microsoft/pylance-release#1351] [Ideas]
When you open a new blank notebook, you're able to run cells immediately but when you need to import in data like a csv from a local file it's confusing to know that you're in a buffer and not the file system yet. And so it's confusing to know which directory you're in to make the path from. 

A simpler solution could be to force the user to save the Notebook somewhere before attempting to run the cell but I'm not sure if that's a better experience.

**Comment by brettcannon** [comment]
If you want to know which directory you're in, [you can use `os.getcwd()` or `os.path.abspath('')`](https://stackoverflow.com/questions/39125532/file-does-not-exist-in-jupyter-notebook).

**Reply by tanhakabir** [reply]
right! but I was wondering if we could have autocompletions so it's easier to debug for those aren't aware of that function

**Reply by brettcannon** [reply]
In that case you would need to have the language server know that you're wanting a file path (which is hard with so many APIs taking a string as a file path). I also don't know how the trigger would work for completions since you don't want every string to be considered a file path.

But ultimately this would be an ask for Pylance, so I'm transferring this to them.

**Reply by jakebailey** [reply]
Trying to find our past conversation about this and failing; this seems like something that Jupyter should already be providing as a part of its completions, as they use the Jupyter + Pylance to build their list, and I think they do path completion. We don't use discussions to track enhancements, so a new issue would be warranted if I can't find our last conversation.
